src/vcpkg/commands.applocal.cpp

```diff
@@ -0,0 +1,538 @@
+            Debug::print("Imported DLLs of ", binary, " were ", Strings::join("\n", imported_names), "\n");
+
+            for (auto&& imported_name : imported_names)
+            {
+                if (m_searched.find(imported_name) != m_searched.end())
+                {
+                    Debug::print("  %s: previously searched - Skip", imported_name);
+                    continue;
+                }
+                m_searched.insert(imported_name);
+
+                Path target_binary_dir = binary.parent_path();
+                Path installed_item_file_path = m_installed_bin_dir / imported_name;
+                Path target_item_file_path = Path(target_binary_dir) / imported_name;
+
+                if (m_fs.exists(installed_item_file_path, VCPKG_LINE_INFO))
+                {
+                    deploy_binary(m_deployment_dir, m_installed_bin_dir, imported_name);
+
+                    if (openni2_installed)
+                    {
+                        deployOpenNI2(target_binary_dir, m_installed_bin_parent, imported_name);
+                    }
+
+                    if (azurekinectsdk_installed)
+                    {
+                        deployAzureKinectSensorSDK(target_binary_dir, m_installed_bin_parent, imported_name);
+                    }
+
+                    if (magnum_installed)
+                    {
+                        bool g_is_debug = m_installed_bin_parent.stem() == "debug";
+                        if (g_is_debug)
+                        {
+                            deployMagnum(target_binary_dir, m_installed_bin_parent / "bin/magnum-d", imported_name);
+                        }
+                        else
+                        {
+                            deployMagnum(target_binary_dir, m_installed_bin_parent / "bin/magnum", imported_name);
+                        }
+                    }
+
+                    if (qt_installed)
+                    {
+                        deployQt(m_deployment_dir, m_installed_bin_parent / "plugins", imported_name);
+                    }
+
+                    resolve(m_deployment_dir / imported_name);
+                }
+                else if (m_fs.exists(target_item_file_path, VCPKG_LINE_INFO))
+                {
+                    Debug::print("  %s: %s not found in %s; locally deployed",
+                                 imported_name,
+                                 imported_name,
+                                 m_installed_bin_parent.c_str());
+                    resolve(installed_item_file_path);
+                }
+                else
+                {
+                    Debug::print("  %s: %s not found", imported_name, installed_item_file_path.c_str());
+                }
+            }
+        }
+
+    private:
+        // Azure kinect sensor SDK plugins
+        void deployAzureKinectSensorSDK(const Path& target_binary_dir,
```

Comment:
I'd like to make a common function instead of these, using a record list file to determine which file should be deployed.
Specific ports should not be associated with vcpkg-tool.

Reply:
I agree, we have this contemplated for the second phase of this project. This ad hoc solution is only for this first phase of my internship. I will start working on the spec for the next phase soon :)